crates/red_knot_python_semantic/src/types.rs

```diff
@@ -320,6 +320,14 @@ impl<'db> Type<'db> {
         matches!(self, Type::Dynamic(DynamicType::Todo(_)))
     }
 
+    pub fn contains_todo(&self, db: &'db dyn Db) -> bool {
+        if let Type::Union(union) = self {
+            union.elements(db).iter().any(Type::is_todo)
+        } else {
+            false
+        }
+    }
```

Comment:
Not sure if this should internally check `self.is_todo()` and short circuit.

Reply:
That is what I'd expected, yeah... a todo type "contains todo"

Reply:
Ok right, I figured and pushed the changes already.